src/hooks/useQuiz.ts

```diff
@@ -1,56 +1,31 @@
 import { UseQuiz } from '../types';
 import useCioClient from './useCioClient';
 import useConsoleErrors from './useConsoleErrors';
-import useQuizApiState from './useQuizApiState';
+import usePropsGetters from './usePropsGetters';
 import useQuizEvents from './useQuizEvents';
-import useQuizLocalState from './useQuizLocalState';
+import useQuizState from './useQuizState';
 
-const useQuiz: UseQuiz = ({
-  quizId,
-  apiKey,
-  cioJsClient,
-  quizVersionId,
-  resultsPageOptions,
-  sessionStateOptions,
-}) => {
-  // Log console errors for required parameters quizId and resultsPageOptions
-  useConsoleErrors(quizId, resultsPageOptions);
+const useQuiz: UseQuiz = (quizOptions) => {
+  const { apiKey, cioJsClient } = quizOptions;
 
-  // Quiz Local state
-  const {
-    quizLocalState,
-    resetQuizLocalState,
-    dispatchLocalState,
-    hydrateQuizLocalState,
-    hasQuizStoredState,
-    resetQuizStoredState,
-  } = useQuizLocalState(sessionStateOptions?.sessionStateKey);
+  // Log console errors for required parameters quizId and resultsPageOptions
+  useConsoleErrors(quizOptions);
 
   // Quiz Cio Client
   const cioClient = useCioClient({ apiKey, cioJsClient });
 
-  // Quiz API state
-  const { isFirstQuestion, quizApiState, resetQuizApiState } = useQuizApiState(
-    quizId,
-    quizLocalState,
-    dispatchLocalState,
-    resultsPageOptions,
-    quizVersionId,
-    cioClient
-  );
+  // Quiz state (Local and API)
+  const quizState = useQuizState(quizOptions, cioClient);
 
   // Quiz callback events
-  const quizEvents = useQuizEvents({
-    cioClient,
-    quizApiState,
-    resultsPageOptions,
-    dispatchLocalState,
-    resetQuizApiState,
-    resetQuizLocalState,
-    hydrateQuizLocalState,
-    resetQuizStoredState,
-    hasQuizStoredState,
-  });
+  const quizEvents = useQuizEvents(quizOptions, cioClient, quizState);
+
+  // Props getters
+  const { quizApiState, quizLocalState } = quizState;
+  const propGetters = usePropsGetters(quizEvents, quizApiState, quizLocalState);
+
+  console.log(quizLocalState.answers);
```

Comment:
Is this intentional?

Reply:
Will remove it before merging. But it's good for debugging now